core/arch/arm/include/kernel/secure_partition.h

```diff
@@ -0,0 +1,168 @@
+#define SP_VER_MAJOR		(0 & SP_VER_MAJOR_MASK)
+#define SP_VER_MINOR		(1 & SP_VER_MINOR_MASK)
+#define SP_VERSION		((SP_VER_MAJOR << 16) | (SP_VER_MINOR))
+
+#define SP_MEM_ATTR_ACCESS_MASK		0x3
+#define SP_MEM_ATTR_ACCESS_NONE		0
+#define SP_MEM_ATTR_ACCESS_RW		1
+#define SP_MEM_ATTR_ACCESS_RO		3
+#define SP_MEM_ATTR_EXEC_NEVER		BIT(2)
+#define SP_MEM_ATTR_EXEC		0
+#define SP_MEM_ATTR_ALL			(SP_MEM_ATTR_ACCESS_RW | \
+					 SP_MEM_ATTR_ACCESS_RO | \
+					 SP_MEM_ATTR_EXEC_NEVER)
+
+/*
+ * Used for EDK2 StMM communication. Since StMM can be launched on an arbitrary
+ * address it uses these 2 syscalls to define the memory attributes for the
+ * data and code segments after dispatching the binaries.
+ *
+ * FFA_SVC_MEMORY_ATTRIBUTES_SET_64:
+ *  - x4: base address
+ *  - x5: number of pages
+ *  - x6: attributes of tha remapping (described above)
+ *
+ * FFA_SVC_MEMORY_ATTRIBUTES_GET_64: currently only a single page is requested
+ *  - x4: base address
+ */
+#define FFA_SVC_MEMORY_ATTRIBUTES_GET_64	UINT32_C(0xC4000064)
+#define FFA_SVC_MEMORY_ATTRIBUTES_SET_64	UINT32_C(0xC4000065)
+/*
+ * We need to define the RPMB IDs formally, since the plan is
+ * for them to be included in the FFA spec (for SP-to-SP future communication).
+ * This is fine for now as it represents the internal contract between the
+ * EDK2 RPMB driver and Secure Partition
+ *
+ * FFA_SVC_RPMB_WRITE:
+ *  - x4: va address of the buffer to write in the device
+ *  - x5: buffer length
+ *  - x6: offset in the device
+ * FFA_SVC_RPMB_READ:
+ *  - x4: va address of the buffer were RPMB contents are copied
+ *  - x5: length to read
+ *  - x6: offset in the device
+ */
+#define FFA_SVC_RPMB_READ		UINT32_C(0xC4000066)
+#define FFA_SVC_RPMB_WRITE		UINT32_C(0xC4000067)
+
+/* Param header types */
+#define SP_PARAM_EP			UINT8_C(0x01)
+#define SP_PARAM_IMAGE_BINARY		UINT8_C(0x02)
+#define SP_PARAM_BL31			UINT8_C(0x03)
+#define SP_PARAM_BL_LOAD_INFO		UINT8_C(0x04)
+#define SP_PARAM_BL_PARAMS		UINT8_C(0x05)
+#define SP_PARAM_PSCI_LIB_ARGS		UINT8_C(0x06)
+#define SP_PARAM_SP_IMAGE_BOOT_INFO	UINT8_C(0x07)
+
+/* Param header version */
+#define SP_PARAM_VERSION_1		UINT8_C(0x01)
+#define SP_PARAM_VERSION_2		UINT8_C(0x02)
+
+/*
+ * This structure provides version information and the size of the
+ * structure, attributes for the structure it represents
+ */
+struct sp_param_header {
+	uint8_t type;		/* type of the structure */
+	uint8_t version;	/* version of this structure */
+	uint16_t size;		/* size of this structure in bytes */
+	uint32_t attr;		/* attributes: unused bits SBZ */
+};
+
+/*
+ * Flags used by the secure_partition_mp_info structure to describe the
+ * characteristics of a cpu. Only a single flag is defined at the moment to
+ * indicate the primary cpu.
+ */
+#define MP_INFO_FLAG_PRIMARY_CPU	UINT32_C(0x00000001)
+
+/*
+ * This structure is used to provide information required to initialise a S-EL0
+ * partition.
+ */
+struct secure_partition_mp_info {
+	uint64_t		mpidr;
```

Comment:
We're usually only having a single space between the type and field name in structs.